src/integrationTest/java/com/mongodb/hibernate/jdbc/MongoStatementIntegrationTests.java

```diff
@@ -244,10 +245,14 @@ private void assertExecuteUpdate(
                             connection.commit();
                         }
                     }
-                    var realDocuments = stmt.getMongoDatabase()
+                    var realDocuments = stmt
+                            .getMongoDatabase()
                             .getCollection("books", BsonDocument.class)
                             .find()
-                            .into(new HashSet<>());
+                            .into(new ArrayList<>())
+                            .stream()
+                            .sorted(Comparator.comparing(doc -> doc.getInt32("_id")))
+                            .toList();
```

Comment:
We should sort in MongoDB (`Sorts` is in `com.mongodb.client.model`):

```suggestion
                            .sort(Sorts.ascending("_id"))
                            .into(new ArrayList<>());
```